big_tests/tests/sasl_external_SUITE.erl

```diff
@@ -85,6 +130,51 @@ cert_with_cn_no_xmpp_addrs_request_name_empty(C) ->
 
     escalus_connection:stop(Client).
 
+self_signed_cert_fails_to_authenticate_with_tls(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_tcp).
+
+self_signed_cert_fails_to_authenticate_with_ws(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_ws).
+
+self_signed_cert_fails_to_authenticate_with_bosh(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_bosh).
+
+self_signed_cert_fails_to_authenticate(C, EscalusTransport) ->
+    Self = self(),
+    F = fun() ->
+		UserSpec = generate_user(C, "alice-self-signed", EscalusTransport),
+		{ok, Client, _} = escalus_connection:start(UserSpec),
+		Self ! escalus_connected,
+		escalus_connection:stop(Client)
+	end,
+    %% We spawn the process trying to connect because otherwise the testcase may crash
+    %% due linked process crash (client's process are started with start_link)
+    Pid = spawn(F),
+    MRef = erlang:monitor(process, Pid),
+
+    receive
+	{'DOWN', MRef, process, Pid, _Reason} ->
+	    ok;
+	escalus_connected ->
+	    ct:fail(authenticated_but_should_not)
+    after 10000 ->
```

Comment:
In case of WS and BOSH there is 5s timeout on connection. The timeout is triggered if the connection cannot be established because of the self-signed cert. This actually inside gun, so we don't have control on it. I added another 5s overhead just because of travis.
I'll change the failure reason.